test/boilerplate_suggestors/boilerplate_utilities_test.dart

```diff
@@ -317,49 +272,101 @@ void main() {
+          semverUtilitiesTestHelper(
+            path: 'test/boilerplate_suggestors/semver_report.json',
+            shouldTreatAllComponentsAsPrivate: true,
+            isValidFilePath: false,
+          );
         });
       });
 
-      test('if props class is in export list', () {
-        final input = '''
+      group('json file does exist', () {
+        semverUtilitiesTestHelper(
+          path: 'test/boilerplate_suggestors/semver_report.json',
+          shouldTreatAllComponentsAsPrivate: false,
+          isValidFilePath: true,
+        );
+      });
+    });
+  });
+}
+
+void semverUtilitiesTestHelper({
+  String path,
+  bool shouldTreatAllComponentsAsPrivate,
+  bool isValidFilePath,
+}) {
+  SemverHelper semverHelper;
+
+  setUpAll(() async {
+    semverHelper = await getSemverHelper(path,
+        shouldTreatAllComponentsAsPrivate: shouldTreatAllComponentsAsPrivate);
+  });
+
+  test('if props class is not in export list', () {
+    final input = '''
+      @Props()
+      class _\$FooProps extends UiProps{
+        String foo;
+        int bar;
+      }
+    ''';
+
+    CompilationUnit unit = parseString(content: input).unit;
+    expect(unit.declarations.whereType<ClassDeclaration>().length, 1);
+
+    unit.declarations.whereType<ClassDeclaration>().forEach((classNode) {
+      expect(
+          semverHelper.getPublicExportLocations(classNode),
+          isValidFilePath || shouldTreatAllComponentsAsPrivate
+              ? isEmpty
+              : [reportNotAvailableComment]);
+      expect(isPublic(classNode, semverHelper),
+          isValidFilePath || shouldTreatAllComponentsAsPrivate ? false : true);
+    });
+  });
+
+  test('if props class is in export list', () {
+    final input = '''
         @Props()
         class ButtonProps extends UiProps{
           String foo;
           int bar;
         }
       ''';
-
-        CompilationUnit unit = parseString(content: input).unit;
-        expect(unit.declarations.whereType<ClassDeclaration>().length, 1);
-
-        unit.declarations.whereType<ClassDeclaration>().forEach((classNode) {
-          expect(semverHelper.getPublicExportLocations(classNode), [
+    final expectedOutput = isValidFilePath
+        ? [
             'lib/web_skin_dart.dart/ButtonProps',
-            'lib/another_file.dart/ButtonProps'
-          ]);
-          expect(isPublic(classNode), true);
-        });
-      });
+            'lib/another_file.dart/ButtonProps',
+          ]
+        : [reportNotAvailableComment];
+
+    CompilationUnit unit = parseString(content: input).unit;
+    expect(unit.declarations.whereType<ClassDeclaration>().length, 1);
+
+    unit.declarations.whereType<ClassDeclaration>().forEach((classNode) {
+      expect(semverHelper.getPublicExportLocations(classNode),
+          shouldTreatAllComponentsAsPrivate ? isEmpty : expectedOutput);
+      expect(isPublic(classNode, semverHelper),
+          shouldTreatAllComponentsAsPrivate ? false : true);
```

Comment:
This could be simplified to:
```suggestion
          !shouldTreatAllComponentsAsPrivate);
```

or, instead of `false` / `true`, the matchers `isFalse` or `isTrue` could be used

Reply:
lol you're very right